tests/test_mixed_simulator_device.py

```diff
@@ -0,0 +1,910 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the MixedStateMixedStateSimulatorDevice
+"""
+import pytest
+import math
+
+import pennylane as qml
+import numpy as np
+from pennylane_cirq import MixedStateSimulatorDevice
+import cirq
+
+
+class TestDeviceIntegration:
+    """Tests that the MixedStateSimulatorDevice integrates well with PennyLane"""
```

Comment:
These are copies of the tests for the pure state simulator

Reply:
Maybe for next time, but I think that these tests could be written much cleaner by reusing the code from `test_simulator_device.py`, since the tests are almost identical.

Reply:
I think @josh146 will disagree with you on this :laughing:

I actually began it the way you suggested (being economical, not principled), but it would have required pretty much every single test to have a conditional branch which determined which device was being used and switched the expected answer. I decided this was a better approach